streams/src/main/java/org/apache/kafka/streams/query/InternalQueryResultUtil.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.query;
+
+import org.apache.kafka.common.annotation.InterfaceStability.Unstable;
+
+import java.util.List;
+
+/**
+ * This utility class is an internal API, but it must be located in the same package as
+ * {@link QueryResult} so that it can access the package-private constructor
```

Comment:
I would prefer to move this class into an internal package and make the constructor of QueryResult `public` -- it won't be useful for KS devs, but it also does not hurt if we allow them to create a `QueryResult`?

Or should we extend `QueryResult` ? (Btw: why is `QueryResult` not an interface?)

Reply:
Thanks!

There was no particular reason not to make it an interface. I went ahead and did so in the latest commit, so you can see if you think it's more elegant that way.

I placed the concrete implementations as well as the util class in an `internal` package, which also means that the special constructor can remain package-private and be accessible from the util.

Reply:
> There was no particular reason not to make it an interface.

I guess, we should switch this around and define everything as an interface by default, and only use classes if there is a good reason for it.